src/DotNetWorker.Core/Context/FunctionContextBindingFeatureExtensions.cs

```diff
@@ -0,0 +1,142 @@
+    {
+        /// <summary>
+        /// Binds an input binding item for the requested <see cref="BindingMetadata"/> instance.
+        /// </summary>
+        /// <param name="context">The function context.</param>
+        /// <param name="bindingMetadata">The BindingMetadata instance for which input data should bound to.</param>
+        /// <returns>An instance of T if binding was successful, else null</returns>
+        public static async ValueTask<T?> BindInputAsync<T>(this FunctionContext context, BindingMetadata bindingMetadata)
+        {
+            if (bindingMetadata == null)
+            {
+                throw new ArgumentNullException(nameof(bindingMetadata));
+            }
+
+            // find the parameter from function definition for the bindingMetadata requested.
+            FunctionParameter? parameter = null;
+            foreach (var param in context.FunctionDefinition.Parameters)
+            {
+                if (param.Name == bindingMetadata.Name)
+                {
+                    parameter = param;
+                    break;
+                }
+            }
+
+            if (parameter == null)
+            {
+                return default;
+            }
+
+            ConversionResult bindingResult;
+            var cacheKey = bindingMetadata.Name;
+            var bindingCache = context.InstanceServices.GetService<IBindingCache<ConversionResult>>();
+            
+            if (bindingCache!.TryGetValue(cacheKey, out var cachedResult))
+            {
+                bindingResult = cachedResult;
+                return (T?)bindingResult.Value;
+            }
+
+            var requestedType = typeof(T);
+            bindingResult = await GetConvertedValueFromInputConversionFeature(context, bindingMetadata, requestedType);
+            bindingCache.TryAdd(cacheKey, bindingResult);
+
+            if (bindingResult.Status == ConversionStatus.Succeeded && bindingResult.Value != null)
+            {
+                return (T?)bindingResult.Value;
+            }
+
+            return default;
+        }
+
+        /// <summary>
+        /// Gets the invocation result of the current function invocation.
+        /// </summary>
+        /// <param name="context">The function context instance.</param>
+        /// <returns>An instance of <see cref="InvocationResult{T}"/>.</returns>
+        /// <exception cref="InvalidOperationException">Throws when the invocation result is not of the requested type.</exception>
+        public static InvocationResult<T> GetInvocationResult<T>(this FunctionContext context)
+        {
+            if (context.GetBindings().InvocationResult is T resultAsT)
+            {
+                return new InvocationResult<T>(context, resultAsT);
+            }
+
+            throw new InvalidOperationException("Invocation result is not of the requested type. Consider using the overload which does not specify the type.");
```

Comment:
nit:
I think leaving the message as `"Invocation result is not of the requested type." is sufficient. One thing that might be helpful is also adding the type of the object assigned to `InvocationResult` here

Reply:
Updated the exception message to include the type of invocation result in the latest iteration.